src/app/components/elements/panels/SelectedEventDetails.tsx

```diff
@@ -1,61 +1,112 @@
-                }
-
-                <strong>Event status: </strong>
-                <span className={selectedEvent.isCancelled ? "text-danger font-weight-bold" : ""}>{selectedEvent.eventStatus}</span>
-                <br />
-
-                <strong>Event start: </strong>
-                <DateString>{selectedEvent.date}</DateString> - <DateString>{selectedEvent.endDate}</DateString>
-                <br />
-
-                <strong>Booking deadline: </strong>
-                <DateString>{selectedEvent.bookingDeadline}</DateString>
-                <br />
-
-                <strong>Prepwork deadline: </strong>
-                <DateString>{selectedEvent.prepWorkDeadline}</DateString>
-                <br />
-
-                {/* Group token is currently JSON Ignored by the API */}
-                {/*<strong>Group Auth Code:</strong>*/}
-                {/*{selectedEvent.isaacGroupToken}*/}
-                {/*<br />*/}
-
-                <span className={zeroOrLess(selectedEvent.placesAvailable) ? "text-danger" : ""}>
-                    <strong>Number of places available: </strong>
-                    {selectedEvent.placesAvailable} / {selectedEvent.numberOfPlaces}
-                </span>
-            </p>}
-        </RS.CardBody>
+import {
+  AppState,
+  getEvent,
+  useAppDispatch,
+  useAppSelector,
+} from "../../../state";
+import { Link } from "react-router-dom";
+import { DateString } from "../DateString";
+import { NOT_FOUND, zeroOrLess } from "../../../services";
+
+export const SelectedEventDetails = ({ eventId }: { eventId: string }) => {
+  const dispatch = useAppDispatch();
+
+  const [studentCount, setStudentCount] = useState(0);
+  const [teacherCount, setTeacherCount] = useState(0);
+
+  useEffect(() => {
+    dispatch(getEvent(eventId));
+  }, [dispatch, eventId]);
+
+  const selectedEvent = useAppSelector((state: AppState) => {
+    return state && state.currentEvent;
+  });
+  const eventBookings = useAppSelector(
+    (state: AppState) => (state && state.eventBookings) || []
+  );
+
+  useEffect(() => {
+    console.log("running rolesOfBookedUsers");
+    let newStudentCount = 0;
+    let newTeacherCount = 0;
+
+    eventBookings.forEach((booking) => {
+      const role = booking.userBooked?.role;
+      if (role === "STUDENT") {
+        newStudentCount++;
+      } else if (role === "TEACHER") {
+        newTeacherCount++;
+      }
+    });
+
+    setStudentCount(newStudentCount);
+    setTeacherCount(newTeacherCount);
+  }, [eventBookings]);
+
+  return (
+    <RS.Card>
+      <RS.CardBody>
+        <h3 className="h-subtitle mb-1">Selected event details</h3>
+        {selectedEvent && selectedEvent !== NOT_FOUND && (
+          <p className="m-0">
+            <strong>Event: </strong>
+            <Link to={`/events/${selectedEvent.id}`} target="_blank">
+              {selectedEvent.title} {selectedEvent.subtitle}
+            </Link>
+            <br />
+            {selectedEvent.location &&
```

Comment:
Here's another good example: rather than having these three conditions, you could move them to a simple function that accepts a location and that returns a boolean. Then you call it like this: `hasValidLocation(selectedEvent) && (React.Fragment>...`. Does that make sense? That makes it a lot easier to read, because the function name tells you its exact intent. By leaving it as it is, I have to make an effort and try to figure out what is the intention of the programmer. Even in this case, which is easy to understand, it's still preferable to state the intention via a function or variable name. It makes your code easier to follow and, therefore, less prone to errors.

I also know that you didn't write this code, so you don't need to change it, but it's an example, and it's important that you're vigilant, because you're by yourself in this project, so it's easy to pick up bad habits.

Another example with this code that I think it would look better would be to move this location logic & fragment to its own component (that should live in this file). It only needs two props (`location` and `isVirtual`). Then you call it like this, for instance:

```typescript
<Location location={selectedEvent.location} isVirtual={isVirtual} />
```

This way you go from 11 lines to 1, and the main component is easier to read.

A good giveaway to know when it's time to refactor is function length. For instance, in this file you only have one single function (`SelectedEventDetails`) that is 100 lines long. That's normally too much.

Again, you don't need to change this as part of this PR, because I think you haven't changed this logic at all, but please have this advice into account when you create new code or modify existing code.